### PR TITLE
Align QueryImage and QueryImageResponse XML with the spec

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
#### Problem
QueryImage is not using a list where it should and QueryImageResponse does not have various fields marked as optional.

#### Change overview
Fix those issues.

#### Testing
Ran manual test of ota-requestor and ota-provider.

@carol-apple @holbrookt Would it make sense to change `HandleQueryImage` to just take the response struct as an arg instead of the individual args?

### CODE REVIEW

**Comment by holbrookt** [comment]
> @carol-apple @holbrookt Would it make sense to change HandleQueryImage to just take the response struct as an arg instead of the individual args?
Yes, we discussed doing this for OTA Requestor cluster as well. I would support that change

**Review by woody-apple — APPROVED** [review]

**Review by holbrookt — APPROVED** [review]
Looks good, just one question about array lengths

**Review by holbrookt — CHANGES_REQUESTED** [review]
Approval was premature :). Please make the changes you suggested - to change the signature of `HandleQueryImage` to take in the command args struct

**Comment by carol-apple** [comment]
> > @carol-apple @holbrookt Would it make sense to change HandleQueryImage to just take the response struct as an arg instead of the individual args?
> 
> Yes, we discussed doing this for OTA Requestor cluster as well. I would support that change

Yes, please!

File: src/app/zap-templates/zcl/data-model/chip/chip-ota.xml [hunk 1]
```diff
@@ -67,13 +67,13 @@ limitations under the License.
...
-            <arg name="updateToken" type="OCTET_STRING" length="32"/>
-            <arg name="userConsentNeeded" type="BOOLEAN" default="false"/>
-            <arg name="metadataForRequestor" type="OCTET_STRING" length="512"/>
+            <arg name="delayedActionTime" type="INT32U" optional="true"/>
+            <arg name="imageURI" type="CHAR_STRING" optional="true"/>
+            <arg name="softwareVersion" type="INT32U" optional="true"/>
+            <arg name="softwareVersionString" type="CHAR_STRING" optional="true"/>
+            <arg name="updateToken" type="OCTET_STRING" length="32" optional="true"/>
+            <arg name="userConsentNeeded" type="BOOLEAN" default="false" optional="true"/>
+            <arg name="metadataForRequestor" type="OCTET_STRING" length="512" optional="true"/>
```
[the remark this answers is not in the document]
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/chip-ota.xml:76]: That "length" field is ignored in practice at the moment. Nothing does anything with it....

**Comment by bzbarsky-apple** [comment]
> Please make the changes you suggested - to change the signature of HandleQueryImage to take in the command args struct

Done, and same for HandleApplyUpdateRequest.  And fixed a few bugs in the caller of that while I was dealing with it anyway.....

File: src/app/clusters/ota-provider/ota-provider-delegate.h [hunk 2]
```diff
@@ -35,15 +37,12 @@ namespace Clusters {
...
-    virtual EmberAfStatus HandleApplyUpdateRequest(CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
-                                                   const chip::ByteSpan & updateToken, uint32_t newVersion) = 0;
+    virtual EmberAfStatus HandleQueryImage(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                           const OtaSoftwareUpdateProvider::Commands::QueryImage::DecodableType & commandData) = 0;
+
+    virtual EmberAfStatus
+    HandleApplyUpdateRequest(CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+                             const OtaSoftwareUpdateProvider::Commands::ApplyUpdateRequest::DecodableType & commandData) = 0;
 
     virtual EmberAfStatus HandleNotifyUpdateApplied(const chip::ByteSpan & updateToken, uint32_t softwareVersion) = 0;
```
Comment (holbrookt) [inline src/app/clusters/ota-provider/ota-provider-delegate.h:47]: Might as well be consistent and make the same change to `HandleNotifyUpdateApplied`, no?

Comment (bzbarsky-apple) [reply src/app/clusters/ota-provider/ota-provider-delegate.h]: Done.

**Comment by bzbarsky-apple** [comment]
@msandstedt @saurabhst @LuDuda @Damian-Nordic @jepenven-silabs @jmartinez-silabs Could you please take a look?

**Review by saurabhst — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by holbrookt — APPROVED** [review]

**Pull request merged**